src/store/fs.rs

```diff
@@ -1188,8 +1328,14 @@ impl FsStore {
 
     /// Load or create a new store with custom options, returning an additional sender for file store specific commands.
     pub async fn load_with_opts(db_path: PathBuf, options: Options) -> anyhow::Result<FsStore> {
+        static THREAD_NR: AtomicU64 = AtomicU64::new(0);
         let rt = tokio::runtime::Builder::new_multi_thread()
-            .thread_name("iroh-blob-store")
+            .thread_name_fn(|| {
+                format!(
+                    "iroh-blob-store-{}",
+                    THREAD_NR.fetch_add(1, Ordering::SeqCst)
```

Comment:
```suggestion
                    THREAD_NR.fetch_add(1, Ordering::Relaxed)
```
No need for `SeqCst` here.